### PR TITLE
feat: add right click events and context menu functionality

### PR DESCRIPTION
**Pull request by marcelimati**
Refferencing: #189 
I resynced branches and re-published changes, should be fine now

### CODE REVIEW

File: projects/extensions/grid/grid.ts [hunk 1]
```diff
@@ -474,7 +483,15 @@ export class MtxGrid implements OnChanges, AfterViewInit, OnDestroy {
       this._toggleNormalCheckbox(rowData);
     }
 
-    this.rowClick.emit({ rowData, index });
+    this.rowClick.emit({ rowData, index, event });
+  }
+
+  /** Row right click event */
+  _rightClickRow(event: MouseEvent, rowData: Record<string, any>, index: number) {
```
Comment (nzbin) [inline projects/extensions/grid/grid.ts:490]: What's the usage scenario of `rightClickRow` event?

File: projects/dev-app/src/app/grid/grid-demo.component.ts [hunk 2]
```diff
@@ -21,10 +21,14 @@ export class GridDemoComponent implements OnInit, AfterViewInit {
   @ViewChild('grid', { static: true }) grid!: MtxGrid;
   @ViewChild('grid2', { static: true }) grid2!: MtxGrid;
 
+  @Output() rowRightClick = new EventEmitter<any>();
```
Comment (nzbin) [inline projects/dev-app/src/app/grid/grid-demo.component.ts:24]: Is this redundant?

File: projects/dev-app/src/app/grid/grid-demo.component.ts [hunk 3]
```diff
@@ -88,7 +92,6 @@ export class GridDemoComponent implements OnInit, AfterViewInit {
       header: this.translate.stream('city'),
       field: 'city',
       minWidth: 150,
-      class: 'text-center',
```
Comment (nzbin) [inline projects/dev-app/src/app/grid/grid-demo.component.ts:91]: Please merge the lastest code.

File: projects/dev-app/src/app/grid/grid-demo.component.ts (hunk 2 above)
Comment (marcelimati) [reply projects/dev-app/src/app/grid/grid-demo.component.ts:24]: Yes, I didn't notice that sorry

File: projects/dev-app/src/app/grid/grid-demo.component.ts (hunk 3 above)
Comment (marcelimati) [reply projects/dev-app/src/app/grid/grid-demo.component.ts:91]: Weird, I forked main branch yesterday and it didn't fetched that change for some reason

**Comment by nzbin** [comment]
Many thanks for your PR. I have achieved this feature with another way.
Please check the context menu example.
https://ng-matero.github.io/extensions/components/grid/overview#context-menu

**Pull request closed without merging**